### ISSUE TITLE
Cannot install golangci-lint because of revive v1.1.3 dependency GOSUM DB error

### ISSUE THREAD

**Issue by jasdel** [golangci/golangci-lint#2575]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Looks like the `github.com/mgechev/revive` dependency rewrote some of their history, and republished release v1.1.3 causing and error trying to install `golangci-lint`.

```
verifying github.com/mgechev/revive@v1.1.3/go.mod: checksum mismatch
        downloaded: h1:ZZq2bmyssGh8MSPz3VVziqRNIMYTJXzP8MUKG90vZ9A=
        go.sum:     h1:jMzDa13teAuv/KLeqgJw79NDe+1IT0ZO3Mht0vN1Yls=
```

The issue was reported to them as https://github.com/mgechev/revive/issues/634 

### Version of golangci-lint

Release [v1.44.0](https://github.com/golangci/golangci-lint/releases/tag/v1.44.0)

Unable to run go install command because `revive` dependency gosum DB issue.

### Configuration file

n/a

### Go environment

<details>

```console
$ go version && go env
go version go1.17.6 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/jasdel/Library/Caches/go-build"
GOENV="/Users/jasdel/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/jasdel/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/jasdel/go"
GOPRIVATE=""
GOPROXY="direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GOVCS=""
GOVERSION="go1.17.6"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/7f/ynvg_sjn7dqc9dxz23x5c8pr0000gs/T/go-build2097961417=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
N/A cannot install
```

</details>

### Code example or link to a public repository

<details>
n/a
</details>

**Comment by jasdel** [comment]
Attempting to use `GOSUMDB=off` will bypass the history rewrite, but get a compile bug when doing so, 



```
GOSUMDB=off go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
...
# github.com/golangci/golangci-lint/pkg/golinters
../../../go/pkg/mod/github.com/golangci/golangci-lint@v1.44.0/pkg/golinters/revive.go:68:22: not enough arguments in call to "github.com/mgechev/revive/lint".New
        have (func(string) ([]byte, error))
        want ("github.com/mgechev/revive/lint".ReadFile, int)
```

**Comment by ldez** [comment]
Hello,

they created a new version to fix the problem https://github.com/mgechev/revive/issues/634#issuecomment-1040881748

**Comment by zhixinwen** [comment]
Can you cut a new release with this fix? I think `go get` still have the error by default.
